go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -481,6 +490,29 @@ func (oc *SecondaryLayer2NetworkController) deleteNodeEvent(node *corev1.Node) e
 	return nil
 }
 
+// addUDNClusterSubnetEgressSNAT adds the SNAT on each node's GR in L2 networks
+// snat eth.dst == 76:23:f 169.254.0.12 10.128.0.0/14
+// snat eth.dst == 76:23:f 169.254.0.12 2010:100:200::/64
+// these SNATs are required for pod2Egress traffic in LGW mode and pod2SameNode traffic in SGW mode to function properly on UDNs
```

Comment:
so this function is just for layer 2 networks right?

Reply:
yes! the ClusterSubnet one is only called for layer2